src/main/webapp/javascript/RecipeRendering.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Takes the recipe info of a recipe as a JS object and the div that the recipe
+ * card will be added to. Creates only one recipe card. For the method to work,
+ * recipeInfo is expected to have the following data saved as properties,
+ * regardless of the explicit property name:
+ * a recipe name, recipe tags,  recipe description, and recipe ID to access more
+ * data on the recipe page.
+ */
+createRecipeCard = (divId, recipeInfo) => {
+    let docDiv = document.getElementById(divId);
+
+    // Creates the recipeDiv container and sets the class name
+    let recipeDiv = document.createElement("div");
+    recipeDiv.setAttribute("class", "recipe-card");
+
+    /**
+     * Creates the html content to include in the recipe card div.
+     * This approach was preferred to conserve space vs using 
+     * document.createElement and other methods 
+     */
+    let htmlString =
+        createElement(recipeInfo["name"], "p", "class=\"recipe-card-name\"") +
+        createElement(recipeInfo["description"], "p", "class=\"recipe-card-description\"");
+
+    recipeDiv.innerHTML = htmlString;
+    docDiv.appendChild(recipeDiv);
+}
+
+/**
+ * Takes an object which can be represented as a string or a string, an html
+ * tag, and the html tag options, to create the html text. Returns the resulting
+ * text.
+ */
+createElement = (object, htmlTag, tagOptions) => {
+    return "<" + htmlTag + " " + tagOptions + ">" + object + "</" + htmlTag + ">";
+}
+
+/**
+ * Method used for testing, currently used in test.jsp
+ * Use this to experiment with any loading functionality.
+ */
+load = () => {
+    createRecipeCard("content", {"name":"test", "description":"this is a test"});
```

Comment:
Do you want to extend this test a little bit? For example, it may be interesting to write some sort of test for this that guarantees a `p` element exists with the value `test` and class `recipe-card-name` and the same for the description.  That way you don't have to manually look at the results to check.

This could be something as simple as printing to the console (for now) or writing test results to the page (in test.jsp).

Additionally, if you write the tests before you refactor `createRecipeCard` to not use innerHTML, you could have the added confidence of knowing your tests still pass after you refactor :)

Reply:
Will do, I'll include a test in my next commit to be checked out.

Reply:
Test file is now included